federated_aws_cli/listener.py

```diff
@@ -70,23 +70,26 @@ def set_role():
 
 @app.route("/api/roles", methods=["GET"])
 def get_roles():
-    roles = []
+    roles = {}
     for arn in login.role_map["roles"]:
         account_id = arn.split(":")[4]
         alias = login.role_map.get("aliases", {}).get(account_id, [account_id])[0]
-        role = arn.split(':')[5].split('/')[-1]
 
-        roles.append(
-            {
-                "alias": alias,
-                "arn": arn,
-                "id": account_id,
-                "role": role,
-            }
-        )
+        role = {
+            "alias": alias,
+            "arn": arn,
+            "id": account_id,
+            "role": arn.split(':')[5].split('/')[-1],
+        }
+
+        if alias in roles:
+            roles[alias].append(role)
+        else:
+            roles[alias] = [role]
 
     # Sort the list by role name
```

Comment:
You probably want to update this comment as we're no longer sorting on role name

Reply:
We are still sorting on role name, just inside the account level. I should actually make a tweak to sort by account name as well.

Reply:
Never mind, I forgot that Flask already sorts anything returned via `jsonify()`.